application/src/main/java/org/opentripplanner/updater/alert/GtfsRealtimeAlertsUpdater.java

```diff
@@ -63,32 +64,28 @@ public String toString() {
   }
 
   @Override
-  protected void runPolling() {
-    try {
-      final FeedMessage feed = otpHttpClient.getAndMap(
-        URI.create(url),
-        this.headers.asMap(),
-        FeedMessage.PARSER::parseFrom
-      );
+  protected void runPolling() throws InterruptedException, ExecutionException {
+    final FeedMessage feed = otpHttpClient.getAndMap(
+      URI.create(url),
+      this.headers.asMap(),
+      FeedMessage.PARSER::parseFrom
+    );
 
-      long feedTimestamp = feed.getHeader().getTimestamp();
-      if (feedTimestamp == lastTimestamp) {
-        LOG.debug("Ignoring feed with a timestamp that has not been updated from {}", url);
-        return;
-      }
-      if (feedTimestamp < lastTimestamp) {
-        LOG.info("Ignoring feed with older than previous timestamp from {}", url);
-        return;
-      }
+    long feedTimestamp = feed.getHeader().getTimestamp();
+    if (feedTimestamp == lastTimestamp) {
+      LOG.debug("Ignoring feed with a timestamp that has not been updated from {}", url);
+      return;
+    }
+    if (feedTimestamp < lastTimestamp) {
+      LOG.info("Ignoring feed with older than previous timestamp from {}", url);
+      return;
+    }
 
-      // Handle update in graph writer runnable
-      saveResultOnGraph.execute(context ->
-        updateHandler.update(feed, context.gtfsRealtimeFuzzyTripMatcher())
-      );
+    // Handle update in graph writer runnable
+    saveResultOnGraph
+      .execute(context -> updateHandler.update(feed, context.gtfsRealtimeFuzzyTripMatcher()))
+      .get();
```

Comment:
I think we should have a timeout here - what would happen if a task does not return ?

Reply:
The task is to write the update into the graph. I am not sure what will happen if the task to update the graph is forcibly terminated.

Reply:
If we don't want to use the asynchronicity of the Future, why do we even need a Future at all? I think in such a case the method could just return void.

In any case, we need to discuss this in the dev meeting.

Reply:
After further discussion I finally realized what's going on here. In fact @leonardehrenfried, this is using the Future to delay completion of the polling graph updater until the subtask it enqueues on a different ExecutorService completes. I will comment further on the main discussion thread.